data/esdt/common.go

```diff
@@ -0,0 +1,84 @@
+package esdt
+
+import "strings"
+
+const (
+	// esdtTickerNumRandChars represents the number of hex-encoded random characters sequence of a ticker
+	esdtTickerNumRandChars = 6
+	// separatorChar represents the character that separated the token ticker by the random sequence
+	separatorChar = "-"
+	// minLengthForTickerName represents the minimum number of characters a token's ticker can have
+	minLengthForTickerName = 3
+	// maxLengthForTickerName represents the maximum number of characters a token's ticker can have
+	maxLengthForTickerName = 10
+	// maxLengthESDTPrefix represents the maximum number of characters a token's prefix can have
+	maxLengthESDTPrefix = 4
+	// minLengthESDTPrefix represents the minimum number of characters a token's prefix can have
+	minLengthESDTPrefix = 1
+)
+
```

Comment:
should we have IsValidToken (as it was before) to understand prefixed tokens as well, not only non-prefixed ones ?

I think it should and it is possible.

It can be done with len of tokenSplit.

Reply:
In mx-chain-go we don't have any function which only does the validation for a simple token, we only have this func(which also does some validations):
`ExtractTokenIDAndNonceFromTokenStorageKey`.

However, that type of function is not being used in other places, so there was no need to create one and move it to core only to be used in the binary node.